### PR TITLE
Refactor BaseBuilder.parse_access_ncfile

### LINKED ISSUE #186: Return type for parse_access_ncfile
# Is your feature request related to a problem? Please describe.

The return out of `BaseBuilder.parse_access_ncfile` is currently a long tuple of assorted file information. This makes it difficult to remember which item is which during debugging & testing.

# Describe the feature you'd like

We should consider if `parse_access_ncfile` should return a `dict` instead. This will have the advantage of making the return more human-legible. (It will also simplify the `parser` functions of the various builders, which effectively all rebuild the tuple return into a `dict` anyway.)

# Describe alternatives you've considered

Nil.

# Additional context

I'm about to add another issue (#187) to suggest an expansion of the testing for builders, which will be made easier by this feature.

### PR DESCRIPTION
**Pull request by charles-turner-1**
Closes #186, and builds on #221 .

### Changes:
Added two new dataclasses to `src/access_nri_intake/source/utils.py`:
- `_VarInfo`: Dataclass that holds information related to a single variable in a given netCDF file: lists of variable names, cell methods, units ,etc, which default to empty lists. It also contains a method to append to these lists (`append_attrs`), and a method to dump its contents to a dictionary (`to_var_info_dict`), which is used in the constructor of `_AccessNCFileInfo`.
- `_AccessNCFileInfo`: Dataclass to hold information from the netCDF file, containing information about all fields in the netCDF, as well as filename, path, etc. It also has a `to_dict` method which dumps the dataclass to a dictionary, following patterns previously used in parser methods of the different builers.

### CODE REVIEW

File: src/access_nri_intake/source/builders.py [hunk 1]
```diff
@@ -286,19 +293,18 @@ def parse_access_ncfile(cls, fname: str, time_dim: str = "time") -> tuple:
 
         Returns
         -------
-        tuple
-            A tuple containing the information parsed from the file
+        output_nc_info: AccessNCFileInfo
```
Comment (marc-white) [inline src/access_nri_intake/source/builders.py:296]: The class is technically `_AccessNCFileInfo` (note underscore) - will not having that break things if we, e.g., auto-generate documentation?

File: src/access_nri_intake/source/builders.py [hunk 2]
```diff
@@ -590,42 +527,18 @@ def parser(cls, file):
...
-                "variable_cell_methods": variable_cell_methods_list,
-                "variable_units": variable_units_list,
-                "filename": filename,
-                "file_id": file_id,
-            }
-
-            return info
+            ncinfo_dict["file_id"] = re.sub(exp_id, "", ncinfo_dict["file_id"]).strip(
+                "_"
+            )
```
Comment (marc-white) [inline src/access_nri_intake/source/builders.py:537]: Not critical, but I think this line-breaking/indentation looks a bit better:

```suggestion
            ncinfo_dict["file_id"] = re.sub(
                exp_id, "", ncinfo_dict["file_id"]
                ).strip("_")
```

File: src/access_nri_intake/source/utils.py [hunk 3]
```diff
@@ -15,6 +17,80 @@ class EmptyFileError(Exception):
...
+class _AccessNCFileInfo:
+    """
+    Holds information about a NetCDF file that is used to create an intake-esm
+    catalog entry.
+
+    ______
+    Notes:
+    Use of both path and filename seems redundant, but constructing filename from
+    the path using a __post_init__ method makes testing more difficult. On balance,
+    more explicit tests are probably more important than the slight redundancy.
```
Comment (marc-white) [inline src/access_nri_intake/source/utils.py:30]: I don't mind a bit of redundancy here - it's a negligible additional thing we need to track that makes life a bit easier (and then only at catalog build time, which is already a ~1hr HPC job).

File: src/access_nri_intake/source/builders.py (hunk 1 above)
Comment (charles-turner-1) [reply src/access_nri_intake/source/builders.py:296]: Good catch

File: src/access_nri_intake/source/utils.py [hunk 4]
```diff
@@ -77,6 +153,8 @@ def get_timeinfo(
         The dataset to parse the time info from
     filename_frequency: str
         Frequency as determined from the filename
+    filename_frequency: str
+        Frequency as determined from the filename
```
Comment (marc-white) [inline src/access_nri_intake/source/utils.py:157]: I think there's been a merge error here...

File: src/access_nri_intake/source/builders.py (hunk 2 above)
Comment (charles-turner-1) [reply src/access_nri_intake/source/builders.py:537]: I agree - the formatter results here are definitely difficult to read. How does this look to you?
```suggestion
ncinfo_dict["file_id"] = re.sub(
                exp_id,
                "",
                ncinfo_dict["file_id"],
            ).strip("_")
```

**Review by marc-white — APPROVED** [review]
Looks all set

**Pull request merged**